4 ricerca

scrivereleonardo opened the pull request:
Closes #4 

Aggiunta pagina che ha la funzione di stampare uno o più utenti , in base ai filtri scelti.
Sono quindi presenti 2 file principali:

filterUsers.js , il quale contiene il codice Javascript 

SearchFilteredUser.html, che ha la funzione di prendere i "filtri scelti dall' utente" e far apparire a schermo i i risultati della stampa

Scrivere committed:
Aggiunta File per la Ricerca degli Utenti

SearchFilteredUser.html:
è la pagina dal quale l' utente selezionera i
filtri per la ricerca, ed è la pagina sulla quale verrà stampato il
risutlato della ricerca

filterUsers.js: Contiene tutto il codice, JS per lo scorrimento
dell'oggetto

Scrivere committed:
Aggiunte 2 funzione

Aggiunta una funzione per ripulire la pagina una volta lanciata una
funzione di stampa , e aggiunta un altra funzione per tornare alla
scelta iniziale

Scrivere committed:
Pulizia codice varia

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+        <input type="text" name="nomeCercato" id="nomeCercato" required> <br>
+        <button class="greenStandardButton" id="mostraUtenti" onclick="searchByName()">Avvia Ricerca</button>
+        <button class="greenStandardButton end" id="indietro" onclick="goBack()">Indietro</button>
+
```
Questo spazio vuoto sarebbe meglio da togliere

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+    <!-- ------------------------------------------------------------------- -->
+
+
+    <!-- Nel caso venisse effettuatua una ricerca PER COMPETENZA , faccio apparire il sottostante -->
```
```suggestion
    <!-- Nel caso venisse effettuata una ricerca PER COMPETENZA , faccio apparire il blocco sottostante -->
```

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+
+    <!-- Nel caso venisse effettuatua una ricerca PER COMPETENZA , faccio apparire il sottostante -->
+    <div id="searchByCompetence" hidden>
+        Inserisci persone con competenza in:
```
```suggestion
        Ricerca persone con competenza in:
```

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+        <input type="text" name="competenzaCercata" id="competenzaCercata" required> <br><br>
+        <button class="greenStandardButton" id="mostraUtenti" onclick="searchByCompetence()">Avvia Ricerca</button>
+        <button class="greenStandardButton end" id="indietro" onclick="goBack()">Indietro</button>
+
```
Questo spazio vuoto sarebbe meglio da togliere

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+    <!-- Nel caso venisse effettuatua una ricerca PER COMPETENZA , faccio apparire il sottostante -->
+    <div id="searchByCompetence" hidden>
+        Inserisci persone con competenza in:
+        <input type="text" name="competenzaCercata" id="competenzaCercata" required> <br><br>
```
Al posto di un inserimento della competenza sarebbe meglio selezionarla

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+    <!-- ------------------------------------------------------------------- -->
+
+
+    <!-- Nel caso venisse effettuatua una ricerca PER COMPETENZA E LIVELLO , faccio apparire il sottostante -->
```
```suggestion
    <!-- Nel caso venisse effettuata una ricerca PER COMPETENZA E LIVELLO , faccio apparire il blocco sottostante -->
```

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+    <!-- Nel caso venisse effettuatua una ricerca PER COMPETENZA E LIVELLO , faccio apparire il sottostante -->
+    <div id="searchByCompetenceAndLevel" hidden>
+
+        Inserisci persone con competenza in:
```
```suggestion
        Ricerca persone con competenza in:
```

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+        <input type="text" name="competenzaCercata" id="competenzaCercataLivello" required> <br><br>
+
+        Che deve essere di livello almeno:
+
```
Togli lo spazio vuoto

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+        Che deve essere di livello almeno:
+
+        <select class="" name="livelli" id="levels">
+
```
Togli lo spazio vuoto

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+            <option value="2livello">Medio</option>
+            <option value="1livello">Basso</option>
+            <option value="0livello">Senza Esperienza </option>
+
```
Togli lo spazio vuoto

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+        </select>
+
+        <br>
+        <button class="greenStandardButton" id="mostraUtenti" onclick="searchByCompetenceAndLevel()">Avvia
```
Se la classe si chiama greenStandardButton perchè poi il bottone viene visualizzato di colore blu?

Wabri commented on SearchFilteredUser.html:
```diff
@@ -0,0 +1,129 @@
+        <button class="greenStandardButton" id="mostraUtenti" onclick="searchByCompetenceAndLevel()">Avvia
+            Ricerca</button>
+
+        <button class="greenStandardButton end" id="indietro" onclick="goBack()">Indietro</button>
```
Se la classe si chiama greenStandardButton perchè poi il bottone viene visualizzato di colore rosso?

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+    }
+];
+
+
```
Sarebbe comodo avere una lista delle aree

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+
+
+function nameORCompetence() {
+    console.log(document.getElementById("initialChoice").value)
```
Non sarebbe comodo avere una funzione che prende in ingresso l'id e ti restituisce l'elemento? Migliorerebbe la leggibilità del codice e ridurrebbe la duplicazione del codice considerando che `documento.getElementById` lo usi in più parti del codice. Pensaci.

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+
+function nameORCompetence() {
+    console.log(document.getElementById("initialChoice").value)
+    if (document.getElementById("initialChoice").value == "name") {
```
Hai considerato la possibiltà di usare uno switch? potrebbe rendere più leggero il carico visivo del codice, in molti casi gli if a cascata sono una pessima scelta architetturale.

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+}
+
+
+let printSpace = document.getElementById("show");
```
Potresti spiegare l'utilizzo di questa variabile

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+function searchByName() {
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        if (users[index].userName.toLowerCase().includes(document.getElementById("nomeCercato").value.toLowerCase())) {
```
estrai da questa condizione una variabile in cui inserisci il risultato booleano

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
+            let searchedCompetence = document.getElementById("competenzaCercata").value.toLowerCase();
+            if (currentCompetence == searchedCompetence) { //che mal di testa sa condizione
```
Togli commenti non significativi

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        if (users[index].userName.toLowerCase().includes(document.getElementById("nomeCercato").value.toLowerCase())) {
+            document.getElementById("show").innerHTML += "<ul><br><li> Nome e Cognome: " + users[index].userName + "</li><br> <li> Competenze: <br> " + printArrayOfObject(users[index].competenceUser) + "</ul><br><br><hr>";
```
Sarebbe visivamente più comodo avere la stringa su più linee:

https://dev-tips.com/javascript/multi-line-strings-in-javascript

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+function searchByCompetence() {
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
```
Estrai dalla condizione la lunghezza, so che sembra una sciocchezza, ma migliora di tantissimo la leggibilità del codice

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
+            let searchedCompetence = document.getElementById("competenzaCercata").value.toLowerCase();
+            if (currentCompetence == searchedCompetence) { //che mal di testa sa condizione
+                document.getElementById("show").innerHTML += "<ul><br><li> Nome e Cognome: " + users[index].userName + "</li><br> <li> Competenze: <br> " + printArrayOfObject(users[index].competenceUser) + "</ul><br><br><hr>";
```
Sarebbe visivamente più comodo avere la stringa su più linee:

https://dev-tips.com/javascript/multi-line-strings-in-javascript

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+            let levelOfCompetence = users[index].competenceUser[index2].level;
+            let searchedLevelOfCompetence = parseInt(document.getElementById("levels").value);
+            if ((currentCompetence == searchedCompetence) && (levelOfCompetence >= searchedLevelOfCompetence)) { 
+                document.getElementById("show").innerHTML += "<ul><br><li> Nome e Cognome: " + users[index].userName + "</li><br> <li> Competenze: <br> " + printArrayOfObject(users[index].competenceUser) + "</ul><br><br><hr>";
```
Sarebbe visivamente più comodo avere la stringa su più linee:

https://dev-tips.com/javascript/multi-line-strings-in-javascript

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
```
Considerando che currentCompetence non lo modificherai mai più potresti trasformarlo in una costante.

Se pensi su larga scala considera che il let occupa più memoria fisica di un const.

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+    for (let index = 0; index < users.length; index++) {
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
+            let searchedCompetence = document.getElementById("competenzaCercata").value.toLowerCase();
```
Considerando che searchedCompetence non lo modificherai mai più potresti trasformarlo in una costante.

Se pensi su larga scala considera che il let occupa più memoria fisica di un const.

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
```
Considerando che currentCompetence non lo modificherai mai più potresti trasformarlo in una costante.

Se pensi su larga scala considera che il let occupa più memoria fisica di un const.

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+    for (let index = 0; index < users.length; index++) {
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
+            let searchedCompetence = document.getElementById("competenzaCercataLivello").value.toLowerCase();
```
Considerando che searchedCompetence non lo modificherai mai più potresti trasformarlo in una costante.

Se pensi su larga scala considera che il let occupa più memoria fisica di un const.

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+        for (let index2 = 0; index2 < users[index].competenceUser.length; index2++) {
+            let currentCompetence = users[index].competenceUser[index2].nameComp.toLowerCase();
+            let searchedCompetence = document.getElementById("competenzaCercataLivello").value.toLowerCase();
+            let levelOfCompetence = users[index].competenceUser[index2].level;
```
Considerando che levelOfcompetence non lo modificherai mai più potresti trasformarlo in una costante.

Se pensi su larga scala considera che il let occupa più memoria fisica di un const.

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+function printArrayOfObject(arrOfobj) {
+    var output = "";
+    for (let index = 0; index < arrOfobj.length; index++) {
+        output += "<br>  <ul><li>Competenza Conosciuta: " + arrOfobj[index].nameComp + "</li><li>   AreaCompetenza: " + arrOfobj[index].area + "</li><li> Livello Competenza:" + arrOfobj[index].level + "</ul><br><br>";
```
Sarebbe visivamente più comodo avere la stringa su più linee:

https://dev-tips.com/javascript/multi-line-strings-in-javascript

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+}
+
+function clear(space) {
+
```
Togli questo spazio bianco

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+function clear(space) {
+
+    space.innerHTML = "";
+
```
Togli questo spazio bianco

Wabri commented on filterUsers.js:
```diff
@@ -0,0 +1,124 @@
+    document.getElementById("searchByCompetenceAndLevel").setAttribute("hidden", "");
+    document.getElementById("searchByCompetence").setAttribute("hidden", "");
+    document.getElementById("searchByName").setAttribute("hidden", "");
+
```
Togli questi spazi bianco

Wabri reviewed (changes requested):
In generale formatterei meglio il codice eliminando le linee vuote lasciando solo quelle che hanno un significato.

Cercherei di usare solo nomi in inglese per variabili e elementi identificativi.

La leggibilità del codice è importante e permette di mantenere il codice in maniera migliore.

Leomorga committed:
Update branch 4-ricerca

Update dei file searchFilteredUsers.html e filterUsers.js.
Aggiunto un temporaneo style.css

Commit:
Merge branch 'master' into 4-ricerca
